### ISSUE TITLE
Update for 0.66

### ISSUE THREAD

**Issue by robmarkcole** [robmarkcole/Hue-sensors-HASS#50]
In 0.66 the hue library changes. Need to modify this custom component so that it hanles auth without relyin on `phue.conf`. Can have the user configure the `url` as was done [in early release](https://github.com/robmarkcole/Hue-sensors-HASS/blob/e5c24ab097b2271f8028f722a522bf8129490626/custom_components/sensor/hue_sensors.py) of this component.

**Comment by mf-social** [comment]
Can't wait to get this back!  I was just starting to rely on it for automations when they changed the ball-park.

Balloob was looking in to integrating the sensors with the current component, would it be better to have a chat with him and go for full integration rather than working in parallel towards the same goal?

Either way, thanks for all your hard work on this component, I've used it loads and can't wait to get the functionality back when you sort it 😎

**Comment by robmarkcole** [comment]
Issue resolved in https://github.com/robmarkcole/Hue-sensors-HASS/tree/development
Note that you need to manually edit the `URL`. The custom component is now decoupled from the Hue bridge component, and only contains a single `hue.py` file. 
Cheers

**Comment by zarthan** [comment]
The update works with 0.66 beta. **A big emphasis** on editing URL.Many thanks

**Comment by zarthan** [comment]
I may have spoken too soon. The hue lights are back but the remote is missing. No initial errors I can see so I have some looking to do.

update: If I put the previous hue.py file back in custom_components along with the updated hue.py in sensors, I have the remote but no lights. Remove the hue.py from custom_components I have lights but no remote. What a predicament. I want the remote. Wife wants lights.

**Comment by robmarkcole** [comment]
PLEASE CLARIFY: You are referring to removing the `hue.py` from the `components` directory - this is the file for the hue **hub**. It is no longer required, please delete. If you have an issue with your lights that is a separate issue and should be reported

**Comment by zarthan** [comment]
I removed the hue.py file from the custom_components directory and added the new hue.py  file to the sensors folder. Update to 0.66.0b3. Restart HA. Lights work just fine, however, no remote. No error messages and I do have debug set for the components.
If I return the hue.py file that I deleted from the custom_components folder and restart HA, I get the remote but no lights.

**Comment by robmarkcole** [comment]
I think someone reported something similar once, may be that the old component is still cached somehow. Think the other guy had to reinstall HA

**Comment by zarthan** [comment]
I will reinstall and see how it goes.

**Comment by mf-social** [comment]
Will you be working towards not needing a developer API key, like before?

**Comment by robmarkcole** [comment]
I expect so

**Comment by mf-social** [comment]
Ace, I'll wait for that then :+1:

**Comment by zarthan** [comment]
I installed a fresh ubuntu server and then set HA up with just the hue lights and your component. I grabbed a fresh copy of the updated hue.py and added the URL to my bridge. I added the hue information to configuration.yaml and also added logger with custom_components.sensor.hue: debug. Lights are there but no remote and no errors.

**Comment by robmarkcole** [comment]
@zarthan glad that is resolved, and I will close this issue and update the master branch. Cheers

**Comment by zarthan** [comment]
@robmarkcole The 0.66 release notes specifically mention Hue and component as being incompatible. I don't know how many people read release notes (at least one from where I stand) but since this is fixed, you might want to get that corrected.
